test/featuretransforms.jl

```diff
@@ -0,0 +1,191 @@
+@testset "FeatureTransforms" begin
+    M1 = [0.0 1.0; 1.0 2.0; -0.5 0.0]
+    M2 = [-2.0 4.0; 3.0 2.0; -1.0 -1.0]
+    M3 = [0.0 1.0; -1.0 0.5; -0.5 0.0]
+    M4 = [0.5 -1.0; -5.0 -2.0; 0.0 1.0]
+    ds = KeyedDataset(
+        flatten([
+            :train => [
+                :load => KeyedArray(M1; time=1:3, loc=[:x, :y]),
+                :price => KeyedArray(M2; time=1:3, id=[:a, :b]),
+            ],
+            :predict => [
+                :load => KeyedArray(M3; time=1:3, loc=[:x, :y]),
+                :price => KeyedArray(M4; time=1:3, id=[:a, :b]),
+            ]
+        ])...
+    )
+
+    @testset "transform" begin
+        @test is_transformable(ds)
+    end
+
+    # TODO: use fake Transforms
+    @testset "apply OneToOne" begin
+        p = Power(2)
+
+        @testset "apply" begin
+            @testset "one feature" begin
+                expected = KeyedDataset(
+                    flatten([
+                        :train => [
+                            :load => KeyedArray(M1; time=1:3, loc=[:x, :y]),
+                            :price => KeyedArray(M2.^2; time=1:3, id=[:a, :b]),
+                        ],
+                        :predict => [
+                            :load => KeyedArray(M3; time=1:3, loc=[:x, :y]),
+                            :price => KeyedArray(M4.^2; time=1:3, id=[:a, :b]),
+                        ]
+                    ])...
+                )
+
+                r = FeatureTransforms.apply(ds, p, (:_, :price, :_))
+
+                @test r isa KeyedDataset
+                @test isequal(r, expected)
+                @test !isequal(ds, expected)
+            end
+
+            @testset "all features" begin
+                expected = KeyedDataset(
+                    flatten([
+                        :train => [
+                            :load => KeyedArray(M1.^2; time=1:3, loc=[:x, :y]),
+                            :price => KeyedArray(M2.^2; time=1:3, id=[:a, :b]),
+                        ],
+                        :predict => [
+                            :load => KeyedArray(M3.^2; time=1:3, loc=[:x, :y]),
+                            :price => KeyedArray(M4.^2; time=1:3, id=[:a, :b]),
+                        ]
+                    ])...
+                )
+
+                r = FeatureTransforms.apply(ds, p; dims=:)
+
+                @test r isa KeyedDataset
+                @test isequal(r, expected)
+                @test !isequal(ds, expected)
+            end
+
+            @testset "inds" begin
```

Comment:
I'm not sure if we have to test inds here as that's part of the FeatureTransforms implementation. It's not affected directly in this package so we should just be able to trust that it works downstream.